flink-streaming-connectors/flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/common/RMQConnectionConfig.java

```diff
@@ -0,0 +1,455 @@
+		return password;
+	}
+
+	/**
+	 * Retrieve the URI.
+	 * @return the connection URI when connecting to the broker
+     */
+	public String getUri() {
+		return uri;
+	}
+
+	/**
+	 * Returns automatic connection recovery interval in milliseconds.
+	 * @return how long will automatic recovery wait before attempting to reconnect, in ms; default is 5000
+	 */
+	public int getNetworkRecoveryInterval() {
+		return networkRecoveryInterval;
+	}
+
+	/**
+	 * Returns true if automatic connection recovery is enabled, false otherwise
+	 * @return true if automatic connection recovery is enabled, false otherwise
+	 */
+	public boolean isAutomaticRecovery() {
+		return automaticRecovery;
+	}
+
+	/**
+	 * Returns true if topology recovery is enabled, false otherwise
+	 * @return true if topology recovery is enabled, false otherwise
+	 */
+	public boolean isTopologyRecovery() {
+		return topologyRecovery;
+	}
+
+	/**
+	 * Retrieve the connection timeout.
+	 * @return the connection timeout, in milliseconds; zero for infinite
+	 */
+	public int getConnectionTimeout() {
+		return connectionTimeout;
+	}
+
+	/**
+	 * Retrieve the requested maximum channel number
+	 * @return the initially requested maximum channel number; zero for unlimited
+	 */
+	public int getRequestedChannelMax() {
+		return requestedChannelMax;
+	}
+
+	/**
+	 * Retrieve the requested maximum frame size
+	 * @return the initially requested maximum frame size, in octets; zero for unlimited
+	 */
+	public int getRequestedFrameMax() {
+		return requestedFrameMax;
+	}
+
+	/**
+	 * Retrieve the requested heartbeat interval.
+	 * @return the initially requested heartbeat interval, in seconds; zero for none
+	 */
+	public int getRequestedHeartbeat() {
+		return requestedHeartbeat;
+	}
+
+	/**
+	 *
+	 * @return Connection Factory for RMQ
+	 * @throws Exception if Malformed URI has been passed
+     */
+	public ConnectionFactory getConnectionFactory() throws Exception {
+		ConnectionFactory factory = new ConnectionFactory();
+		if (this.uri != null && !this.uri.isEmpty()){
+			try {
+				factory.setUri(getUri());
+			}catch (Exception e){
+				LOG.error("Failed to parse uri {}", e.getMessage());
+				throw e;
+			}
+		} else {
+			factory.setHost(getHost());
+			factory.setPort(getPort());
+			factory.setVirtualHost(getVirtualHost());
+			factory.setUsername(getUsername());
+			factory.setPassword(getPassword());
+		}
+
+		factory.setAutomaticRecoveryEnabled(isAutomaticRecovery());
+		factory.setConnectionTimeout(getConnectionTimeout());
+		factory.setNetworkRecoveryInterval(getNetworkRecoveryInterval());
+		factory.setRequestedHeartbeat(getRequestedHeartbeat());
+		factory.setTopologyRecoveryEnabled(isTopologyRecovery());
+		factory.setRequestedChannelMax(getRequestedChannelMax());
+		factory.setRequestedFrameMax(getRequestedFrameMax());
+
+		return factory;
+	}
+
+	public static class Builder {
+		/** The default host */
+		public static final String DEFAULT_HOST = "localhost";
```

Comment:
Okay, I didn't check all the fields in the ConnectionFactory, sorry.

Your proposal sounds good.